pkg/core/transactor/transactor.go

```diff
@@ -163,6 +158,11 @@ func (t *Transactor) Bid(ctx context.Context, c *node.BidRequest) (*node.Transac
 
 // Stake will create a staking transaction.
 func (t *Transactor) Stake(ctx context.Context, c *node.StakeRequest) (*node.TransactionResponse, error) {
+	// Are we synced?
+	if !t.canStake() {
+		return nil, errors.New("node is not synced")
+	}
+
```

Comment:
I like it. Failing early before pushing requests in the channel is a good strategy

Reply:
Make sure to add this to the development guidelines